### PR TITLE
Bug 724648 amo api for auth

### PR DESCRIPTION
**Pull request by zalun**
removed all calls for AMO database
- Show old auth only if switch is on
  this is for devs to be able to log-in without AMO API
- removed authentication against database in AMOAuthentication
- get_amo_cursor removed

### CODE REVIEW

File: apps/amo/authentication.py [hunk 1]
```diff
@@ -48,46 +40,12 @@ def authenticate(self, username, password):
...
-
-        # Manage profile
-        try:
-            profile = user.get_profile()
-        except Profile.DoesNotExist:
-            profile = Profile(user=user)
-
-        profile.update_from_AMO(self.user_data)
-        return user
+        return None
```
Comment (seanmonstar) [inline apps/amo/authentication.py:43]: `authenticate` is supposed to return `None` no matter what? It's not supposed to return `user` if one was found?

Comment (zalun) [inline apps/amo/authentication.py:43]: isn't it used by old-style login only?

Comment (seanmonstar) [inline apps/amo/authentication.py:43]: Hmm, I guess so...

How do we login on local machines, now?

Comment (zalun) [inline apps/amo/authentication.py:43]: That's the beginning of this function: https://github.com/zalun/FlightDeck/blob/81cb8cbf7153d6a45d5af4db0e7cf491baa9a4b1/apps/amo/authentication.py#L37

Comment (seanmonstar) [inline apps/amo/authentication.py:43]: ah, good call. r+

Comment (zalun) [inline apps/amo/authentication.py:43]: merging, we would need to add the 'show-old-login' switch as well

**Pull request merged**